src/components/AddNewCharacterForm/AddNewCharacterForm.js

```diff
@@ -0,0 +1,233 @@
+import { Button, Container, Grid, makeStyles } from "@material-ui/core"
+import { useForm } from "react-hook-form"
+import { useDispatch } from "react-redux"
+import { nanoid } from "@reduxjs/toolkit"
+
+import { addCharacter } from "@/reducers/characterReducer"
+
+import ControlledInputField from "../ControlledInputField"
+import ControlledSelectInputField from "../ControlledSelectInputField/ControlledSelectInputField"
+
+const useStyles = makeStyles((theme) => ({
+  title: {
+    borderBottom: "1px solid grey",
+    padding: theme.spacing(1),
+    fontWeight: theme.typography.fontWeightBold,
+  },
+
+  body: {
+    padding: "15px",
+    display: "flex",
+    flexDirection: "column",
+    gap: theme.spacing(2),
+    width: "30vw",
+  },
+
+  buttonBox: {
+    display: "flex",
+    gap: theme.spacing(2),
+    position: "absolute",
+    bottom: 0,
+    padding: theme.spacing(2),
+    "& > *": {
+      color: theme.palette.common.white,
+    },
+    "& > :first-child": {
+      backgroundColor: theme.palette.button.main,
+    },
+    "& > :last-child": {
+      backgroundColor: theme.palette.grey[400],
+    },
+  },
+}))
+
+const AddNewCharacterForm = ({ setOpen }) => {
+  const classes = useStyles()
+  const { control, handleSubmit } = useForm({
+    defaultValues: {
+      name: "",
+      eyeColor: "",
+      height: "",
+      gender: "",
+      birthYear: "",
+      homeworld: "",
+      species: "",
+      numberOfFilms: "",
+    },
+    reValidateMode: "onSubmit",
+  })
+
+  const dispatch = useDispatch()
+
+  const onSubmit = (data) => {
+    if (data.name && data.eyeColor) {
+      dispatch(
+        addCharacter({
+          id: nanoid(),
+          name: data.name,
+          eyeColor: data.eyeColor,
+          height: data.height,
+          gender: data.gender,
+          birthYear: data.birthYear,
+          homeworld: {
+            id: nanoid(),
+            name: data.homeworld,
+          },
+          species: {
+            id: nanoid(),
+            name: data.species,
+          },
+          filmConnection: {
+            totalCount: data.numberOfFilms,
+          },
+        })
+      )
+    }
+  }
+
+  return (
+    <>
+      <Container
+        className={classes.title}
+        color="text.disabled"
+      >
+        Add New Character
+      </Container>
+      <div className={classes.body}>
```

Comment:
I don't see you handling overflow anywhere in the form, what happens if the size of our form grows and we have many fields in it?